lib/elixir/lib/keyword.ex

```diff
@@ -449,9 +449,20 @@ defmodule Keyword do
   """
   @spec delete(t, key, value) :: t
   def delete(keywords, key, value) when is_list(keywords) and is_atom(key) do
-    :lists.filter(fn {k, v} -> k != key or v != value end, keywords)
+    do_delete(keywords, key, value, _deleted? = false)
```

Comment:
@michalmuskala we're trying to get rid of all current occurrences of `do_` prefixing, and not to introduce new ones. Perhaps `delete_key_value` (`delete_key` for the other function) would be better name?